src/client.rs

```diff
@@ -1,11 +1,22 @@
 use crate::events::serenity_handler::SerenityHandler;
+use serenity::{
+    model::{channel::Message, id::GuildId},
+    prelude::{RwLock, TypeMapKey},
+};
 use songbird::SerenityInit;
-use std::{env, error::Error};
+use std::{collections::HashMap, env, error::Error, sync::Arc};
 
 pub struct Client {
     client: serenity::Client,
 }
 
+pub struct GuildQueueInteractions;
+type QueueMessage = (Message, Arc<RwLock<usize>>);
+
+impl TypeMapKey for GuildQueueInteractions {
+    type Value = HashMap<GuildId, Vec<QueueMessage>>;
+}
+
```

Comment:
Move to its own module. In fact, wait for the approval and merge of https://github.com/aquelemiguel/parrot/pull/98 and let's merge this with the guild settings we have (possibly even rename to `GuildCache`)